pkg/host-disk/host-disk_test.go

```diff
@@ -148,19 +127,25 @@ var _ = Describe("HostDisk", func() {
 		Context("With multiple HostDisk volumes", func() {
 			Context("With non existing disk.img", func() {
 				It("Should create disk.img if there is enough space", func() {
-					By("Creating a new minimal vmi")
-					vmi := api.NewMinimalVMI("fake-vmi")
-
-					By("Adding a HostDisk volumes")
-					addHostDisk(vmi, "volume1", v1.HostDiskExistsOrCreate, "64Mi")
-					addHostDisk(vmi, "volume2", v1.HostDiskExistsOrCreate, "128Mi")
-					addHostDisk(vmi, "volume3", v1.HostDiskExistsOrCreate, "80Mi")
+					By("Creating a new VMI with HostDisk volumes")
+					vmi := libvmi.New(
+						libvmi.WithHostDisk("volume1", createHostDiskPath("volume1"), v1.HostDiskExistsOrCreate),
+						libvmi.WithHostDisk("volume2", createHostDiskPath("volume2"), v1.HostDiskExistsOrCreate),
+						libvmi.WithHostDisk("volume3", createHostDiskPath("volume3"), v1.HostDiskExistsOrCreate),
+					)
+					By("Setting the capacity for each HostDisk volume")
+					vmi.Spec.Volumes[0].HostDisk.Capacity = resource.MustParse("64Mi")
```

Comment:
Same here we shouldn't directly access the volumes directly but instead create an Option in libvmi

Reply:
What about making the size an argument on `createHostDisk` for now?

Reply:
I’ve made some changes to enhance the reusability and flexibility of HostDisk related functions. Here’s a summary of the changes:
1. Created a new function WithHostDiskAndCapacity in `pkg/libvmi/storage.go`
This function allows specifying the capacity directly when creating a HostDisk:
```go
func WithHostDiskAndCapacity(diskName, path string, diskType v1.HostDiskType, capacity string) Option {
    return func(vmi *v1.VirtualMachineInstance) {
        addDisk(vmi, newDisk(diskName, v1.DiskBusVirtio))
        addVolume(vmi, newHostDisk(diskName, path, diskType, resource.MustParse(capacity)))
    }
}
```
2. Modified the `newHostDisk` function to accept an optional capacity argument
This makes newHostDisk more versatile, allowing it to handle both cases with or without specified capacity:
```go
func newHostDisk(name, path string, diskType v1.HostDiskType, capacity ...resource.Quantity) v1.Volume {
    hostDisk := v1.HostDisk{
        Path: path,
        Type: diskType,
    }

    // Set capacity if provided, otherwise default to defaultDiskSize
    if diskType == v1.HostDiskExistsOrCreate {
        if len(capacity) > 0 {
            hostDisk.Capacity = capacity[0]
        } else {
            hostDisk.Capacity = resource.MustParse(defaultDiskSize)
        }
    }

    return v1.Volume{
        Name: name,
        VolumeSource: v1.VolumeSource{
            HostDisk: &hostDisk,
        },
    }
}
```
and we can use it like this
```go
vmi := libvmi.New(
    libvmi.WithHostDiskAndCapacity("volume1", createHostDisk("volume1"), v1.HostDiskExistsOrCreate, "64Mi"),
    libvmi.WithHostDiskAndCapacity("volume2", createHostDisk("volume2"), v1.HostDiskExistsOrCreate, "128Mi"),
    libvmi.WithHostDiskAndCapacity("volume3", createHostDisk("volume3"), v1.HostDiskExistsOrCreate, "80Mi"),
)
```

Reply:
Looks like you missed to update L129-L138